contracts/prize-strategy/multiple-winners/MultipleWinners.sol

```diff
@@ -126,57 +126,72 @@ contract MultipleWinners is PeriodicPrizeStrategy {
+        // Split percentage must be below 1000 and greater then 0 (e.x. 200 is equal to 20% percent)
+        // The range from 0 to 1000 is used for single decimal precision (e.x. 15 is 1.5%)
+        require(split.percentage > 0 && split.percentage <= 1000, "MultipleWinners/invalid-prizesplit-percentage");
 
-            _tempTotalPercentage = _tempTotalPercentage.add(split.percentage);
+        if(_prizeSplits.length <= index) {
+          _prizeSplits.push(split);
+        } else {
+          _prizeSplits[index] = split;
+        }
 
-            emit PrizeSplitSet(split.target, split.percentage, split.token);
+      emit PrizeSplitSet(split.target, split.percentage, split.token, index);
 
-            _prizeSplits.push(split);
-        }
+      _tempTotalPercentage = _tempTotalPercentage.add(split.percentage);
+    }
+
+    // If updating remove outdated prize split configurations not passed in the new prizeStrategySplit 
+    while (_prizeSplits.length > prizeStrategySplit.length) {
+      uint8 _index = _prizeSplits.length.sub(1).toUint8();
+      MultipleWinnersPrizeSplit memory _split = _prizeSplits[_index];
+      delete _prizeSplits[_index];
+      _prizeSplits.pop();
+      emit PrizeSplitRemoved(_split.target, _index);
     }
 
     // The total of all prize splits can NOT exceed 100% of the awarded prize amount.
     require(_tempTotalPercentage <= 1000, "MultipleWinners/invalid-prizesplit-percentage-total");
-
-    // Store the total prize split percentage. Used when adding a single prize split config.
-    _totalPrizeSplitPercentage = _tempTotalPercentage.toUint16();
   }
 
-  /**
+   /**
     * @notice Remove a prize split config 
     * @dev Remove a prize split by passing the index of target prize split config.
     * @param prizeSplitIndex The target index to delete.
   */
-  function removePrizeSplit(uint8 prizeSplitIndex) external onlyOwner {
-    require(prizeSplitIndex <= _prizeSplits.length, "MultipleWinners/invalid-prize-split-index");
-
-     MultipleWinnersPrizeSplit memory removedPrizeSplit = _prizeSplits[prizeSplitIndex];
+  function setPrizeSplit(MultipleWinnersPrizeSplit memory prizeStrategySplit, uint8 prizeSplitIndex) external onlyOwner {
+    require(prizeStrategySplit.target != address(0), "MultipleWinners/invalid-prizesplit-target");
+    require(prizeStrategySplit.percentage > 0 && prizeStrategySplit.percentage <= 1000, "MultipleWinners/invalid-prizesplit-percentage-amount");
 
-    // Maintaining the order of the prizeSplits array is not required. To reduce gas costs the last
-    // prize split config is saved to the index being removed and then deleted from the last array position.
-    // This removes the requirement to shift each item in the array once a configration has been removed.
+    if(prizeSplitIndex > _prizeSplits.length.sub(1)) {
+      // Add a new prize split config to _prizeSplits array.
+      _prizeSplits.push(prizeStrategySplit);
```

Comment:
hmm...I think the function should revert if the user passes an index out-of-bounds

Imagine if I tried to set index 100; it'll push it on but then emit an event with the wrong index.

I think this function should be kept as simple as possible.

Reply:
It will fail if the user passes 100 index. Solidity won't allow updates to a non-existent array element. Can only be set if the array is fixed length.

This current approach allows users to update an existing prize split config or add a new one to the end of array.

This was added to limit gas costs for adding a single prize split config, instead of requiring passing the full length array in `setPrizeSplits`

Reply:
Via the Discord conversation removed the ability to push prize split config in `setPrizeSplit` and relies entirely on `setPrizeSplits` for adding prize splits configs and comparing current vs new prizeSplits before updating the existing prize split config.